[IMP] pos: auto course allocation new section (#18414)
odoo/documentation: content/applications/sales/point_of_sale/restaurant.rst

click :guilabel:`Save`.

To configure courses and assign them to POS product categories, follow these steps:

#. Go to :menuselection:`Point of Sale --> Configuration --> POS Product Categories` and select a
   product category (e.g., *Burgers*).
#. In the :guilabel:`Course` field, select an existing course or type a new one (e.g., *Main*) and
   click :guilabel:`Create`.

Once configured, adding a product from the *Burgers* category to an order automatically assigns it
to the *Main* course.

.. note::
   - A single course can be assigned to multiple POS product categories. For example, the *Main*
     course can be used for the *Burgers*, *Pizzas*, and *Meat* categories.
   - You can create as many courses as necessary.

.. tip::
   By default, courses appear in the cart in the order they were created. To rearrange the course
   order to reflect your service flow, activate the :ref:`developer mode
   <developer-mode/activation>`, go to :menuselection:`Point of Sale --> Configuration --> Courses`,
   then use the :icon:`oi-draggable` :guilabel:`(draggable)` icon to move courses to the desired
   position.

Fire courses
~~~~~~~~~~~~

> Once the order is complete, click :guilabel:`Send` to transmit it to the kitchen. The first course

Reviewer: I was struggling to find the Send button, and then I realised that sometimes, the 'Send' doesn't appear:

<img width="578" height="854" alt="Image" src="https://github.com/user-attachments/assets/bc3d1472-8c84-411f-a907-d8a5fcb00dfb" />

Is there a better way we could express this to avoid confusion?

Author: I could maybe add a note (?) I'll see what I can do :D

appears on the :doc:`preparation display <extra/preparation>` as waiting to be prepared, while

> subsequent courses are marked as :guilabel:`Pending`.

Reviewer: I think I need your help to understand this. For example, on this screenshot, I just sent orders for table 10 to the kitchen. I see the 'pending' for course 2, but I'm not seeing any kind of label on course 1 ... just the little clock icon:

<img width="1502" height="939" alt="Image" src="https://github.com/user-attachments/assets/750c78de-dc0c-488f-aa79-9168d620d9b6" />

Is that what you mean by 'waiting to be prepared'? Or am I missing something?

Author: Exactly, the tag appears only when the order is on hold, not when it needs to be prepared. I'll try to make it clearer

click :guilabel:`Save`.

To configure courses and assign them to POS product categories, follow these steps:

#. Go to :menuselection:`Point of Sale --> Configuration --> POS Product Categories` and select a
   product category (e.g., *Burgers*).
#. In the :guilabel:`Course` field, select an existing course or type a new one (e.g., *Main*) and
   click :guilabel:`Create`.

Once configured, adding a product from the *Burgers* category to an order automatically assigns it
to the *Main* course.

.. note::
   - A single course can be assigned to multiple POS product categories. For example, the *Main*
     course can be used for the *Burgers*, *Pizzas*, and *Meat* categories.
   - You can create as many courses as necessary.

.. tip::
   By default, courses appear in the cart in the order they were created. To rearrange the course
   order to reflect your service flow, activate the :ref:`developer mode
   <developer-mode/activation>`, go to :menuselection:`Point of Sale --> Configuration --> Courses`,
   then use the :icon:`oi-draggable` :guilabel:`(draggable)` icon to move courses to the desired
   position.

Fire courses
~~~~~~~~~~~~

> Once an order is complete, click :guilabel:`Send` to transmit it to the kitchen. The first course

Reviewer: Since the _Send_ label can be hidden (and thank you for the note, that makes it super clear!!), maybe it would be good to also mention where it is? e.g.: 

```suggestion
Once an order is complete, click :guilabel:`Send` at the bottom left of the register to transmit it to the kitchen. The first course
```

.. seealso::
   `Courses (video tutorial) <https://youtu.be/5W2S9HwSrDQ?si=Pauk9gyuypl4NU-M>`_

Automatic course allocation
---------------------------

> The automatic course allocation feature creates courses and automatically sorts products by course.

Reviewer: What do you mean by 'creates courses'? Becuase it seems like the actual creation of defined courses is not dependent on the feature being activated, right?

Maybe:

```suggestion
The automatic course allocation feature allows products added to an order to be automatically assigned to a default course.
```

Author: I meant something like what you understood. It's just that instead of having to click the "Course" button to split an order into courses, it creates them in the cart and automatically sorts products. Without the feature, you have to click the button for courses to appear in the cart.

Author: I'll try to be clearer

To activate and configure the feature:

#. Go to :menuselection:`Point of Sale --> Configuration --> Settings`, scroll down to the

>    :guilabel:`Preparation` section, and enable :guilabel:`Automatic course allocation`.

Reviewer: I'm wondering if it would be better to separate the 'activate' and 'configure'? Right now, both are bundled together, so the sentence that comes after the steps feels a bit odd.

```suggestion
To activate the feature, go to :menuselection:`Point of Sale --> Configuration --> Settings`, scroll down to the :guilabel:`Preparation` section, enable :guilabel:`Automatic course allocation`, then click :guilabel:`Save`.
```
Then, I would start the next part with something like:

```suggestion
   To configure courses and assign product categories to courses, follow these steps:.
```

Question though: is this considered the main way to configure courses? (i.e., not  to do it with debug mode activated, via Configuration > Courses)

Author: Configuration > Courses is only available in debug mode, unfortunately, so yes, it is the main way to configure courses (to assign a course to a pos product category).
I mention debug mode because it's the only way we could find to reorganize courses if the aperitif was created after the dessert and we need it first (for example).

#. Go to :menuselection:`Point of Sale --> Configuration --> POS Product Categories` and select a

>    product category.

Reviewer: Could it help to give an example here? I know I'm new to POS, but I find it a bit abstract the way it is. Maybe something like:

```suggestion
#. Go to :menuselection:`Point of Sale --> Configuration --> POS Product Categories` and select a
   product category, e.g., Burgers.
```

#. Enter a course in the :guilabel:`Course` field and click :guilabel:`Create`, or select an

>    existing course.

Reviewer: ```suggestion
#. Enter a course in the :guilabel:`Course` field, e.g., :guilabel:`Main`, and click :guilabel:`Create`, or select an existing course.
```

Do we need to say something about repeating as many time as needed?

Author: I'm not sure it was necessary, but I've added it as a note along with something I originally didn't think about: the fact that one course can be set for multiple pos product categories, e.g., *Main* could be used for *Burgers*, *Pizzas*, and *Meat*.

When you select any products from that POS category, they are automatically sorted into the

> defined courses.

Reviewer: ```suggestion
Thereafter, when a product from that POS category is added to an order, it is automatically allocated to the defined course.
```

What happens if a customer wants to receive a product with a different course to the one it's automatically assigned to? Like Jane Doe prefers to have the shrimp croquettes (configured as starter) as her main? _Edited to add: just like when manually assigning courses. Which definitely confirms to me that we need to find a way to have a comprehensive Courses section that inclues manual and automatic options_

Also, what happens if there are product categories that do not have a course assigned? Just tried, and they seem to just get added to the previous course 'triggered' ... e.g., if you ordered a burger (Main) and a chocolate mousse (Dessert) and then another customer in the group orders a pizza (no course), then the pizza would end up in the Dessert course.

Author: That's right, they are added to the selected/triggered course. Drinks are a good example because one same drink could be ordered at any moment.
They can transfer any product from one course to another. I'll record a video: https://docs.google.com/videos/d/11b_q_GNegi7NnalW7jwpZ8ZWC5mgByJroYIgqtBm-6I/edit?usp=drive_link

Author: Better yet, click "Main" to trigger it before selecting a pizza, but if the waiter forgot to trigger another course, he can transfer the product to the main course

By default, courses are organized by creation sequence (the first created appears first). If you
create a new course later but want it to appear before existing ones, you must manually change the
sequence order.

To rearrange the course order:

#. Activate :ref:`developer mode <developer-mode/activation>`.
#. Navigate to :menuselection:`Point of Sale --> Configuration --> Courses`.
#. Rearrange the courses by dragging and dropping them into the desired order.

> 

Reviewer: Could this be better as tip? And if yes, condensed a bit? Something like:

```suggestion
By default, courses appear in the register in the order in which they were created. To rearrange the course order to reflect your service flow, activate :ref:`developer mode <developer-mode/activation>`, go to :menuselection:`Point of Sale --> Configuration --> Courses`, then use the :icon:`oi-draggable` :guilabel:`(draggable)` to move courses to the desired position.

```

Author: It's something that made me quite mad when I couldn't find how to rearrange them last time, which is why I thought it could be important to mention outside of a tip, but it actually could be just a tip
